Learning_test_automation/Spotkanie6_Zadania/CurrencyConversion_vBeta/Option/Transaction.cs

```diff
@@ -40,83 +41,91 @@ private MakingDecision OwnCurrency()
                 Console.Write($"I understand my dear user, you have chosen {userCurrency.currencyName}, and how many {userCurrency.currencyName} do you need: ");
 
                 decimal amount = option.GetAmount();
-                userCurrency.UserTransaction = amount;
-
-                Console.WriteLine($"For {amount} {userCurrency.currencyName} you will get: {userCurrency.BoughtValue} PLN.");
+                userCurrency.CurrencyExchange = amount;
 
+                Console.WriteLine($"For {amount} {userCurrency.currencyName} you will get: {userCurrency.CurrencyExchange} PLN.");
+                            
                 return exchangeRate;
             }
             return exchangeRate;
         }
 
-        private void SellCurrency()
-        {        
-            Console.Write("\nDear user, tell me what currency you want to sell: \n");
-            
-            CurrencyDefault currencyDefault = currencyPresentation.GetCurrencyName();
-            string result = currencyDefault.ToString() == "Czech_crown" ? "Czech crown" : currencyDefault.ToString();
-
-            Console.Write($"I understand my dear user, you have chosen {result}, and how many {result}s do you need sell: ");
-            
-            decimal amount = option.GetAmount();
-            string currencySymbol = "None!";
-            decimal sentValue = 0;
-
-            if (currencyDefault == CurrencyDefault.Dollar)
-            {
-                dollar.SentValue = amount;
-                sentValue = dollar.SentValue;
-                currencySymbol = dollar.SymbolCurrency;
-            }
-            else if (currencyDefault == CurrencyDefault.Euro)
+        private void BuyOrSellCurrency()
+        {
+            string caller = new StackTrace().GetFrame(1).GetMethod().Name;
+            bool ifBuy;
+            string transaction;
+            if (caller == "Buying")
             {
-                euro.SentValue = amount;
-                sentValue = euro.SentValue;
-                currencySymbol = euro.SymbolCurrency;
+                ifBuy = true;
+                transaction = "buy";
             }
-            else if (currencyDefault == CurrencyDefault.Czech_crown)
+            else
             {
-                czechCrown.SentValue = amount;
-                sentValue = czechCrown.SentValue;
-                currencySymbol = czechCrown.SymbolCurrency;
+                ifBuy = false;
+                transaction = "sell";
             }
```

Comment:
Logika od linii 55 do 67 jest nawet ciekawa i moglaby znajdowac sie w osobnej metodzie odpowiedzialnej za przypisywanie tych poszczegolnych wartosci.
Dodalbym prywatne pola w klasie `Transcation` dla ktorych wartosci beda ustawiane.

Reply:
kurde zgubiłem to, jeszcze raz musisz mi wskazać.

Reply:
![image](https://user-images.githubusercontent.com/11249681/236840339-6a98d2fe-8d99-400c-b246-1e6b90586a24.png)

Reply:
tego już nie ma xD

Reply:
do poprawy ale metoda OwnCurrency()

Reply:
done.